src/components/BalanceSheet/BalanceSheet.tsx

```diff
@@ -50,33 +54,17 @@ const BalanceSheetView = ({
   withExpandAllButton = true,
   asWidget = false,
   stringOverrides,
+  syncWithGlobalDate,
+  customDateProvider,
 }: BalanceSheetViewProps) => {
   const [effectiveDate, setEffectiveDate] = useState(startOfDay(new Date()))
-  const { data, isLoading, refetch } = useBalanceSheet(effectiveDate)
+  const { data, isLoading } = useBalanceSheet(effectiveDate)
 
   const [view, setView] = useState<ViewType>('desktop')
   const containerRef = useElementViewSize<HTMLDivElement>(newView =>
     setView(newView),
   )
 
-  useEffect(() => {
-    // Refetch only if selected effectiveDate and data's effectiveDate are different
-    const d1 =
-      effectiveDate &&
-      format(startOfDay(effectiveDate), 'yyyy-MM-dd\'T\'HH:mm:ssXXX')
-    const d2 =
-      data?.effective_date &&
-      format(
-        startOfDay(
-          parse(data.effective_date, 'yyyy-MM-dd\'T\'HH:mm:ssXXX', new Date()),
-        ),
-        'yyyy-MM-dd\'T\'HH:mm:ssXXX',
-      )
-    if (d1 && (!d2 || (d2 && d2 !== d1))) {
-      refetch()
-    }
-  }, [effectiveDate])
-
```

Comment:
Clean up: no needed and it could actually double requests because `useSWR` send request due new key based on the dates change, and then removed `refetch()` could repeat same request.